### PR TITLE
fix(vcs): fixed the upload vcs page

### LINKED ISSUE #111: Code is getting broken on uploading files by VCS
### Description

On giving the url to repository on VCS page, it is making request to the `upload/{id}` but then the code gets broke.

#### How to reproduce

- Go to `/uploads/vcs`
- Try uploading the files.

### PR DESCRIPTION
**Pull request by Shruti3004**
## Description

Vcs page is getting broken on making the request.

### Changes

- Changes the promise rejection parameter.
- Change the retries left to the same property in sendrequest.
- Return the function if retries are greater than 0.

## How to test

Visit to `/upload/vcs`
 Closes #111

### CODE REVIEW

File: src/pages/Upload/Vcs/index.jsx [hunk 1]
```diff
@@ -88,7 +88,7 @@ const UploadFromVcs = () => {
   let uploadId;
 
   // Data required for creating the upload
-  const [uploadVcsData, setUploadFileData] = useState(initialState);
+  const [uploadVcsData, setUploadVcsData] = useState(initialState);
```
Comment (GMishx) [inline src/pages/Upload/Vcs/index.jsx:91]: Can you please explain this change? Its bit confusing to me.

Comment (Shruti3004) [reply src/pages/Upload/Vcs/index.jsx:91]: Actually I checked that variable name is not appropriate, for it we are doing upload by vcs, so its change nothing technically but just more readable.

Comment (GMishx) [reply src/pages/Upload/Vcs/index.jsx:91]: Ah sorry, I missed the `set`. I thought its `[uploadVcsData, uploadVcsData]` :no_mouth:

**Review by GMishx — APPROVED** [review]
Changes looks good. Tested, working as expected.

**Pull request merged**